proxy/agent_test.go

```diff
@@ -304,3 +304,39 @@ func TestRecovery(t *testing.T) {
 	<-time.After(100 * time.Millisecond)
 	close(stop)
 }
+
+// TestCascadingAbort plays a scenario that may trigger self-abort deadlock:
+//  * start three epochs 0, 1, 2
+//  * epoch 2 crashes, triggers abort of 0, 1
+//  * epoch 1 crashes before receiving abort, triggers abort of 0
+//  * epoch 0 aborts, recovery to config 2 follows
+func TestCascadingAbort(t *testing.T) {
```

Comment:
Were you able to reliably reproduce the deadlock with this test case prior to the proxy/agent.go fixes?